Sources/PackageRegistryTool/PackageRegistryTool+Publish.swift

```diff
@@ -0,0 +1,270 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import Basics
+import Commands
+import CoreCommands
+import Foundation
+import PackageModel
+import PackageRegistry
+import TSCBasic
+import Workspace
+
+extension SwiftPackageRegistryTool {
+    struct Publish: SwiftCommand {
+        static let metadataFilename = "package-metadata.json"
+
+        static let configuration = CommandConfiguration(
+            abstract: "Publish to a registry"
+        )
+
+        @OptionGroup(visibility: .hidden)
+        var globalOptions: GlobalOptions
+
+        @Option(name: .customLong("id"), help: "The package identity")
+        var packageIdentity: PackageIdentity
+
+        @Option(help: "The package version")
+        var version: Version
+
+        @Option(name: .customLong("url"), help: "The registry URL")
+        var registryURL: URL?
+
+        @Option(
+            name: .customLong("output-directory"),
+            help: "The path of the directory where output file(s) will be written"
+        )
+        var customWorkingDirectory: AbsolutePath?
+
+        @Option(
+            name: .customLong("metadata-path"),
+            help: "The path to the package metadata JSON file"
+        )
+        var customMetadataPath: AbsolutePath?
+
+        @Option(help: "Signature format identifier. Defaults to 'cms-1.0.0'.")
+        var signatureFormat: SignatureFormat = .CMS_1_0_0
+
+        @Option(help: "The label of the signing identity to be retrieved from the system's secrets store if supported")
+        var signingIdentity: String?
+
+        @Option(help: "The path to the certificate's PKCS#8 private key (DER-encoded)")
+        var privateKeyPath: AbsolutePath?
+
+        @Option(
+            help: "Paths to all of the certificates (DER-encoded) in the chain. The certificate used for signing must be listed first and the root certificate last."
+        )
```

Comment:
nit: line length
```suggestion
        @Option(help: 
            """
            Paths to all of the certificates (DER-encoded) in the chain. The certificate used for signing must be listed \
            first and the root certificate last.
            """
        )
```

Reply:
hmm, I ran the formatter... is it part of the rules?

Reply:
Yes it is, maybe it's not smart enough to reformat single-line strings into multi-line strings.